scripts/check-test-determinism.py

```
@@ -316,6 +356,237 @@ def _is_assertion_line(line: str) -> bool:
 
+def _python_f_string_starts_at(line: str, quote_index: int) -> bool:
+    return bool(_PYTHON_F_STRING_PREFIX.search(line[:quote_index]))
+
+
+def _quote_delimiter_at(line: str, quote_index: int) -> str:
+    quote = line[quote_index]
+    triple = quote * 3
+    return triple if line.startswith(triple, quote_index) else quote
+
+
+def _executable_code_positions(line: str) -> list[bool]:
```

Ruff (0.16.1)

[warning] Too many branches (21 > 12)

(PLR0912)